src/socketio.quiver.js

```diff
@@ -663,41 +740,66 @@ QuiverSocialProvider.prototype.selfDescriptionChanged_ = function() {
     var connections = this.clientConnections_[userId];
     for (var i = 0; i < connections.length; ++i) {
       var connection = connections[i];
-      var introMsg = this.makeIntroMsg_();
-      connection.socket.emit('emit', {
-        'room': userId,
-        'msg': introMsg
-      });
+      var friend = this.configuration_.friends[userId];
+      var introMsg = this.makeIntroMsg_(friend);
+      this.emitEncrypted_(connection.socket, friend, introMsg);
     }
   }
 };
 
 /**
- * @param {string} networkData
+ * @param {Socket} socket
+ * @param {QuiverSocialProvider.userDesc_} friend
+ * @private
+ */
+QuiverSocialProvider.prototype.addDisconnectMessage_ =
+    function(socket, friend) {
+  var disconnectMessage = {
+    'cmd': 'disconnected',
+    'fromClient': this.clientSuffix_
+  };
+  this.emitEncrypted_(socket, friend, disconnectMessage, true);
+};
+
+/**
+ * @param {*} networkData
  * @param {string} inviteResponse
- * @param {Function} cb
+ * @param {!Function} cb
  * @override
  */
 QuiverSocialProvider.prototype.acceptUserInvitation = function(networkData, inviteResponse, cb) {
-  var invite = /** QuiverSocialProvider.invite_ */ JSON.parse(networkData);
-  this.addFriend_(invite.servers, invite.userId, invite.nick, inviteResponse, cb);
+  var invite = /** @type {!QuiverSocialProvider.invite_} */ (networkData);
```

Comment:
You can remove inviteResponse, now that freedom-social-quiver has knockCodes and encryption again